loki/analyse/util_polyhedron.py

```diff
@@ -0,0 +1,328 @@
+# (C) Copyright 2018- ECMWF.
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+import numpy as np
+
+from loki import TypedSymbol, Loop
```

Comment:
Thanks, I haven't had a chance, yet, to read and review this properly - will try to do this today. But just as a quick response to your question about the cyclic import: This is the place where this is introduced.

Within Loki, you should use fully-qualified imports  wherever possible to avoid these (i.e., here use `sym.TypedSymbol` because the module is already imported, and `from loki.ir import Loop` for the loop node). The reason is that the `__init__.py` imports the entire Loki namespace for ease of use from the outside.

Reply:
Thank you. Fixed in 78422d3